File: Data Structures & Algorithms/find-duplicate-integer/submission-3.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            n = abs(nums[i])
            if nums[n - 1] < 0:
                return abs(nums[i])
            nums[n - 1] = -1 * nums[n - 1]

File: Data Structures & Algorithms/find-pivot-index/submission-3.java
class Solution {
    public int pivotIndex(int[] nums) {
        int totalSum = 0;
        for (int n: nums){
            totalSum += n;
        }
        
        int leftSideSum = 0;
        for(int i = 0; i < nums.length; i++) {
            int rightSideSum = totalSum - leftSideSum - nums[i];
            if (leftSideSum == rightSideSum) {
                return i;
            }
            leftSideSum += nums[i];
        }

        return -1;
    }
}